packages/semcom-node/lib/store/services/component-in-memory-store.service.ts

```diff
@@ -7,16 +8,27 @@ export class ComponentInMemoryStore extends ComponentStore {
   }
 
   async query(filter: Partial<ComponentMetadata>): Promise<ComponentMetadata[]> {
-    return this.components.filter((component) =>
+    let filtered = this.components.filter((component) =>
```

Comment:
I would prefer this to remain a constant, e.g. by returning other constant based on this one.